packages/lodestar/src/eth1/impl/ethers.ts

```diff
@@ -70,40 +70,63 @@ export class EthersEth1Notifier extends (EventEmitter as { new(): Eth1EventEmitt
       this.logger.verbose("Eth1 notifier is disabled" );
       return;
     }
-    if (this.started) {
-      this.logger.verbose("Eth1 notifier already started" );
+    await this.doStart();
+  }
+
+  public async stop(): Promise<void> {
+    this.provider.removeAllListeners("block");
+    // stop processing eth1
+    this.startedProcessEth1 = false;
+    this.logger.verbose("Eth1 notifier stopped");
+  }
+
+  /**
+   * This is triggered when building genesis or after chain gets started.
+   * @param subscribe true for genesis builder
+   * @returns a pushable for genesis builder, null otherwise.
+   */
+  public async startProcessEth1Blocks(subscribe = false): Promise<Pushable<[IDepositEvent[], ethers.providers.Block]>> {
+    if (this.startedProcessEth1) {
+      this.logger.info("Started processing eth1 blocks already");
```

Comment:
I think it should be fine if two modules get this source

Reply:
For now, we consume eth1 block + events either before genesis (for building genesis state), or after genesis (for storing proposing data), not both